navigator_helpers/pipelines/base.py

```diff
@@ -40,8 +40,8 @@ def save_artifacts(self, path: str | Path):
         path = Path(path)
         path.mkdir(parents=True, exist_ok=True)
         self.dataframe.to_json(path / "synthetic_dataset.json", orient="records")
-        with open(path / "metadata.json", "w") as f:
-            json.dump(self.metadata.model_dump(), f)
+        # with open(path / "metadata.json", "w") as f:
+        #     json.dump(self.metadata.model_dump(), f)
```

Comment:
```suggestion
        with open(path / "metadata.json", "w") as f:
            json.dump(self.metadata, f)
```
I think this should work. The `model_dump` is for pydantic objects.

Reply:
I temporary commented this out since it gives me this issue:

```python
Traceback (most recent call last):
  File "/Users/maarten/workspace/gretelai/navigator-helpers/examples/nl2code_pipeline/run_from_yaml.py", line 20, in <module>
    run_pipeline(
  File "/Users/maarten/workspace/gretelai/navigator-helpers/examples/nl2code_pipeline/run_from_yaml.py", line 9, in run_pipeline
    results.save_artifacts("outputs/")
  File "/Users/maarten/venv/lib/python3.11/site-packages/navigator_helpers/pipelines/base.py", line 44, in save_artifacts
    json.dump(self.metadata.model_dump(), f)
              ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'model_dump'
```